src/browser.ts

```diff
@@ -66,6 +66,47 @@ async function runCommand(
 export interface BrowserOptions {
   headless: boolean;
   product: "chrome" | "firefox";
+  userDataDir: string | null;
+  persistent: boolean;
+  gracefulShutdown: boolean;
```

Comment:
I don't really understand what usecase this solves. At most, this PR should introduce the `userDataDir` (kind of wordy name btw) option.

If the user specifies where they want the data dir to be, they probably want it to be persisted. It'd be weird to care about the location and then not make it persisted.

I think `gracefulShutdown` should be true 100% of the time. I don't understand why it's a flag to be enabled.

Reply:
> If the user specifies where they want the data dir to be, they probably want it to be persisted. It'd be weird to care about the location and then not make it persisted.

When I wrote this functionality I was thinking about the possibility of specifying any folder, including its ability to be temporary.

Of course, I could rewrite some of the code and remove the `persistent` option, and only save the directory when `userDataDir` is specified.

> I think gracefulShutdown should be true 100% of the time. I don't understand why it's a flag to be enabled.

Maybe then add an env check if this built-in functionality should be disabled?

```ts
if (Deno.env.get("DISABLE_ASTRAL_GRACEFUL_SHUTDOWN") === undefined) { ... }
```

For example, if you need to implement custom behavior with signals other than the current one and the ability to close all browsers using the `closeAllBrowsers` method.

Reply:
> Of course, I could rewrite some of the code and remove the persistent option, and only save the directory when userDataDir is specified.

I would be in favor of that.

> Maybe then add an env check if this built-in functionality should be disabled?

I don't think it's necessary. If someone asks for it, we can consider it but I think this over-complicates the API surface.

Reply:
> I don't think it's necessary. If someone asks for it, we can consider it but I think this over-complicates the API surface.

In my pet project I have implemented waiting for an empty task queue and disconnecting from the database at termination.

The problem is that signals are processed synchronously and do not wait for other listeners, so I need to disable listening inside this implementation and manually close browsers.

Reply:
If you attempt to terminate a process early, graceful shutdown ensures an orderly and controlled termination of an application, allowing it to complete necessary tasks, release resources, and maintain system stability.